### ISSUE TITLE
Show login form and redo action if successful login when trying to comment on an issue where session has timed out.

### ISSUE THREAD

**Issue by Floaf** [go-gitea/gitea#17282]
### Feature Description

I am having a few issues open in tabs in gitea that I am working on and i write comments to them from time to time. Sometimes it has been a few days in between and the session has expired.
As of now, there is no way of knowing that (that might be a feature request on it's own) so when i write a comment and press "Comment", I am redirected to the startpage and that's it. It is not so user friendly.

I have to login in and then i have to go to my issue again and then write the comment again.

My suggestion is that when you are trying to do an action when session has expired, show the login form and if the user enter the correct credentials, try to redo the action the user was trying to do, like in my case, try to post in the comment again. Then there is no loss of data (and time).

**Comment by Floaf** [comment]
Yes, I am sorry I did not complete the issue form and you are right, I did not have the latest version. But now I just upgraded to version 1.15.4 and i see that I am redirected to the root page if this happens. It is a little bit better but the main problem still exists. All the data that is entered when trying to submit in an expired session is still lost.
My suggestion is not to have a redirect, but to show a login form and on that page capture the submitted post data from the original submit (in hidden form fields?) so if you login again, the original request is performed again and no data is lost.
